### PR TITLE
chore: enable jsdoc/informative-docs internally

### LINKED ISSUE #8156: Repo: Bring in eslint-plugin-jsdoc informative-docs rule
### Suggestion

Splitting out of #7855: Also of interest is [`jsdoc/informative-docs`](https://github.com/gajus/eslint-plugin-jsdoc/blob/aefceca7413ac0263ed1b58127fff4efca89de4f/docs/rules/informative-docs.md). IMO we should enable this to help folks accustomed to writing comments like `@param id the id` (it happens sometimes).

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #8156
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Sending as a draft for reference in triage.

### CODE REVIEW

File: packages/eslint-plugin/src/rules/adjacent-overload-signatures.ts [hunk 1]
```diff
@@ -128,7 +128,6 @@ export default createRule({
     }
 
     /**
-     * Check the body for overload methods.
      * @param {ASTNode} node the body to be inspected.
```
Comment (Josh-Cena) [inline packages/eslint-plugin/src/rules/adjacent-overload-signatures.ts:131]: TBH this is useless too...

File: packages/eslint-plugin/src/rules/member-ordering.ts [hunk 2]
```diff
@@ -632,7 +632,6 @@ function getRank(
  * public-instance-method.
  * If a lowest possible rank is a member group, a comma separated list of ranks is returned.
  * @param ranks the existing ranks in the object.
- * @param target the target rank.
```
Comment (Josh-Cena) [inline packages/eslint-plugin/src/rules/member-ordering.ts:635]: This doesn't look very useless to me? And to me omitting docs for one parameter looks like a greater evil...

File: packages/eslint-plugin/src/rules/space-before-function-paren.ts [hunk 3]
```diff
@@ -70,7 +70,6 @@ export default createRule<Options, MessageIds>({
 
     /**
      * Determines whether a function has a name.
-     * @param {ASTNode} node The function node.
      * @returns {boolean} Whether the function has a name.
```
Comment (Josh-Cena) [inline packages/eslint-plugin/src/rules/space-before-function-paren.ts:73]: Remove this too

**Comment by JoshuaKGoldberg** [comment]
I'll wait to continue -let alone merge- these changes until the flat config work lands.

**Review by bradzacher — APPROVED** [review]

**Pull request merged**